main.cpp

```diff
@@ -14,11 +14,47 @@ Create a branch named Part2
     you should be able to deduce the return type of those functions based on their usage in Person::run()
     You'll need to insert the Person struct from the video in the space below.
  */
+struct Person	
+{	
+	int age;
+	int height;
+	float hairLength;
+	float GPA;
+	unsigned int SATScore;
 
+    struct Foot				
+	{			
+        float distanceFromGround = 0 ;
+        float momentum = 0;
 
+    	int stepForward( float momentum, float distanceFromGround );		
+		float stepSize();		
+	};			
 
+    Foot leftFoot;
+    Foot rightFoot;
 
+    void run( int howFast, bool startWithLeftFoot );
+};
+
+
+int Person::Foot::stepForward( float momentumT0, float distanceFromGroundT0 )				
+{			
+    if ( momentumT0*distanceFromGroundT0 > 0 )
+    
+    return 1;			
+    
+    else return 0 ;
+}			
 
+float Person::Foot::stepSize()				
+{			
+	int legLength = 2;			
+	float sineOfStepAngle = .25;	
+    int stepDirection = Person::Foot::stepForward( momentum, distanceFromGround );
```

Comment:
this is a `Foot` member function.  you can only call it with an instance of `Foot`.  you do not need to fully qualify `stepForward` when you call it.
- [ ] revise

Reply:
after modifying stepforward back to return type void, I could not use stepforward as part of the formula, then I replaced the expresion with 1.